test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuilldIntegrationTestFixture.cs

```diff
@@ -195,6 +195,30 @@ private void DeleteFiles(string destinationDir)
             Directory.Delete(destinationDir, true);
         }
 
+        private static string FindMostRecentPackNupkg(string nupkgDirectory)
+        {
+            var listOfMatchingNupkgs = Directory.GetFiles(nupkgDirectory, "NuGet.Build.Tasks.Pack.*", SearchOption.TopDirectoryOnly)
```

Comment:
This method: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/LocalFolderUtility.cs#L49

Would do all the parsing for you, filter out symbols packages, parse the id and version, and make it easy to sort by version or file time.

Not critical for test code, but it does everything you are doing here.